pkg/registry/common/updatepath/common.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// Copyright (c) 2022 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package updatepath provides a chain element that sets the id of an incoming or outgoing request
+package updatepath
+
+import (
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/common/grpcmetadata"
+)
+
+/*
+Logic for Update path:
+
+ 0. if Index == 0, and there is no current segment will add one.
+ 1. If current path segment.Name is equal to segmentName passed, it will just update current connection.Id and exit.
+ 2. If current path segment.Name is not equal to segmentName:
+    2.0 if current path segment.Id is not equal to current connection.Id, will return error.
+    2.1 if path has next segment available, but next name is not equal to segmentName, will update both next name and connection.Id.
+    2.2 if no next path segment available, it will add one more path segment and generate new Id, update connection.Id.
+    2.3 if path has next segment available and next name is segmentName, take Id from next path segment.
```

Comment:
Please double check this description (as I see it was taken from `networkservice`). Or add more comments to the code below.